text/0000-nested-publish.md

```diff
@@ -0,0 +1,359 @@
+- Feature Name: `nested_publish`
+- Start Date: 2023-06-30
+- RFC PR: [rust-lang/rfcs#3452](https://github.com/rust-lang/rfcs/pull/3452)
+- Rust Issue: ...
+
+# Summary
+[summary]: #summary
+
+Allow published Cargo packages to depend on other packages stored within themselves, as is currently possible in unpublished packages.
+
+# Motivation
+[motivation]: #motivation
+
+There are a number of reasons why a Rust developer currently may feel the need to create multiple library crates, and therefore multiple Cargo packages (since one package contains at most one library crate). These multiple libraries could be:
+
+* A trait declaration and a corresponding derive macro (which must be defined in a separate proc-macro library).
```

Comment:
Should `-sys` packages be a motivation with using this being a recommended path or should we discourage using this with `-sys` 

This was talked about a little at https://github.com/rust-lang/rfcs/pull/2224#issuecomment-348053490